plugin/parser/engine/pg/deparse.go

```diff
@@ -44,10 +44,140 @@ func deparse(context parser.DeparseContext, in ast.Node, buf *strings.Builder) e
 			return err
 		}
 		return buf.WriteByte(';')
+	case *ast.CreateIndexStmt:
+		if err := deparseCreateIndex(context, node, buf); err != nil {
+			return err
+		}
+		return buf.WriteByte(';')
 	}
 	return errors.Errorf("failed to deparse %T", in)
 }
 
+func deparseCreateIndex(context parser.DeparseContext, in *ast.CreateIndexStmt, buf *strings.Builder) error {
+	if err := context.WriteIndent(buf, parser.DeparseIndentString); err != nil {
+		return err
+	}
+
+	if _, err := buf.WriteString("CREATE "); err != nil {
+		return err
+	}
+
+	if in.Index.Unique {
+		if _, err := buf.WriteString("UNIQUE "); err != nil {
+			return err
+		}
+	}
+
+	if _, err := buf.WriteString("INDEX "); err != nil {
+		return err
+	}
+
+	if in.IfNotExists {
+		if _, err := buf.WriteString("IF NOT EXISTS "); err != nil {
+			return err
+		}
+	}
+
+	if err := writeSurrounding(buf, in.Index.Name, `"`); err != nil {
+		return err
+	}
+
+	if _, err := buf.WriteString(" ON "); err != nil {
+		return err
+	}
+
+	if err := deparseTableDef(context, in.Index.Table, buf); err != nil {
+		return err
+	}
+
+	if _, err := buf.WriteString(" USING "); err != nil {
+		return err
+	}
+
+	method := ""
+	switch in.Index.Method {
+	case ast.IndexMethodTypeBTree:
```

Comment:
Instead of doing convertion here and above, let's add method convertor for the type IndexMethodType.